src/node_report_utils.cc

```diff
@@ -0,0 +1,491 @@
+ ******************************************************************************/
+void ProcessNodeReportDirectory(const char* args) {
+  if (strlen(args) == 0) {
+    std::cerr << "Missing argument for node-report directory option\n";
+    return;
+  }
+  if (strlen(args) > NR_MAXPATH) {
+    std::cerr << "Supplied node-report directory path too long (max "
+              << NR_MAXPATH
+              << " characters)\n";
+    return;
+  }
+  snprintf(report_directory, sizeof(report_directory), "%s", args);
+}
+
+/*******************************************************************************
+ * Function to process node-report config: verbose mode switch.
+ ******************************************************************************/
+unsigned int ProcessNodeReportVerboseSwitch(const char* args) {
+  if (strlen(args) == 0) {
+    std::cerr << "Missing argument for node-report verbose switch option\n";
+    return 0;
+  }
+  // Parse the supplied switch
+  if (!strncmp(args, "yes", sizeof("yes") - 1) ||
+      !strncmp(args, "true", sizeof("true") - 1)) {
+    return 1;
+  } else if (!strncmp(args, "no", sizeof("no") - 1) ||
+             !strncmp(args, "false", sizeof("false") - 1)) {
+    return 0;
+  } else {
+    std::cerr << "Unrecognised argument for node-report verbose switch option: "
+              << args
+              << "\n";
+  }
+  return 0;  // Default is verbose mode off
+}
+
+/*******************************************************************************
+ * Function to obtain the node and subcomponent version strings.
+ *******************************************************************************/
+std::string GetComponentVersions() {
+  std::string versions = "(";
+  versions += "http_parser: ";
+  versions += NODE_STRINGIFY(HTTP_PARSER_VERSION_MAJOR)
+                            "."
+                            NODE_STRINGIFY(HTTP_PARSER_VERSION_MINOR)
+                            "."
+                            NODE_STRINGIFY(HTTP_PARSER_VERSION_PATCH);
+  versions += ", ";
+  versions += "v8: ";
+  versions += V8::GetVersion();
+
+  versions += ", ";
+  versions += "uv: ";
+  versions += uv_version_string();
+
+  versions += ", ";
+  versions += "zlib: ";
+  versions += ZLIB_VERSION;
+
+  versions += ", ";
+  versions += "ares: ";
+  versions += ARES_VERSION_STR;
+
+  // probably one liner, cut it here.
+  versions += ",\n ";
+  versions += "modules: ";
+  versions += NODE_STRINGIFY(NODE_MODULE_VERSION);
+  versions += ", ";
+  versions += "nghttp2: ";
+  versions += NGHTTP2_VERSION;
+  versions += ", ";
+  versions += "napi: ";
+  versions += NODE_STRINGIFY(NAPI_VERSION);
+
+  // There is additional work required to get the major.minor.patch
+  // format from the OPENSSL_VERSION_TEXT, but as we are generating
+  // a human-readable format, this should be ok for the node report.
+  versions += ", ";
+  versions += "openssl: ";
+  versions += OPENSSL_VERSION_TEXT;
```

Comment:
Will probably need an update to the tests if the format no longer matches that in `process.versions`.

Reply:
sure, will fix it. one test already started to show failure due to this.

Reply:
subsequent reviews and changes thereon made this change un-necessary.